wordpress-mobile/WordPress-Android
# My Site Dashboard-Phase 2: Implement pull-to-refresh

Parent #15215 

This is the final PR in a series (see parent issue)  that refactored, prepared, and implemented end-to-end pull to refresh for the My Site tab. The PTR logic is controlled by the MySiteDashboardPhase2FeatureConfig feature flag, thus minimizing risk to existing functionality.

This PR focuses on the following:
- Ensure that `refresh()` is only called for sources that implement `MySiteRefreshSource`
- Hide the refresh indicator when all sources have finished refreshing
- Kick off `refresh()` when a pull-to-refresh occurs
- Kick off `refresh()` during `onResume` event except for initial request


### **To test:**
#### **Test 1**
- Launch the app
- Navigate to My Site -> App Settings -> Debug Settings
- Ensure that the `MySiteDashboardPhase2FeatureConfig` is to set to off
- Restart the app if needed
- Navigate to the My Site tab
- Try to swiping down on the view to trigger the refresh
- Note: swipe to refresh is not active


#### **Test 2**
- Launch the app
- Navigate to My Site -> App Settings -> Debug Settings
- Set the `MySiteDashboardPhase2FeatureConfig` flag to on
- Restart the App
- Navigate to the My Site tab
- Try to swiping down on the view to trigger the refresh
- Note: 
      The swipe to refresh indicator is shown at first
      The "upcoming scheduled posts" posts cards data will change to say "refreshed" ( they are hardcoded to change)
      Thee  swipe to refresh indicator is no longer shown when all data has reloaded

## Regression Notes
1. Potential unintended areas of impact
The cards don't refresh when requested

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and new automated tests

3. What automated tests I added (or what prevented me from doing so)
New tests in MySiteViewModelTest

PR submission checklist:
